### ISSUE TITLE
Minor usability thing - can we not begin with genitals?

### ISSUE THREAD

**Issue by Vidyut** [makehumancommunity/mpfb2#185]
So this keeps happening. Open the shader editor, begin editing, see no difference. Oops, I'm editing the genitals - an area that has never featured in a single scene I've ever made, and yet it is the most edited shader node of my MPFB2 life - completely unintentionally. Given that the face/body is the most likely to be edited area, is there a way for that group to appear by default/first/active in the vertices or areas or whatever they are called?

On a possibly unrelated note, I haven't been able to use the advanced shader at all on a new laptop with a proper GPU (so it handles everything else much better) RTX 3050. Blender crashes, session crashes, I have to login again. But I'm also using the new Ubuntu 24.04 fresh install so this could be related to something I forgot to install, etc.

I wouldn't even mind deleting the unnecessary code for genitals altogether to save some code processing in case that helps lighten the memory load.

On a humorous note, I'm sure there's a Freudian joke in there somewhere to be staring at a character's face and tweaking genitals... 

And this is going to be totally embarrassing if this is happening because of something I did. ROFL.

**Comment by Vidyut** [comment]
Also on a completely unrelated note, once I'm done making my promo, I can give you the characters I made to use whatever is useful - skins, most likely. It's going slow, because I got a ton of other stuff going on in parallel. I have the skin textures and such - that gets uploaded - but I don't know how to make the hair, wrinkle maps, assets, effects, etc work in uploads like that - stuff that goes with that character. I can share blend files and they can be scavenged into assets by cleverer people for whatever is useful?

There's no hurry to figure this out. You're planning to implement the new hair system anyway so that could happen out of the box and my promo will probably happen long after the book has sunk. lol. I have a few characters ready, but don't want to share before that since I've made so much effort, so I want them to be seen first in the promo.

(they aren't that great - I'm a newbie)

**Comment by joepal1976** [comment]
I suspect that the genitals vertices are pre-selected now as that is the last vertex group to be created. There's no reason for it to remain selected once the character creation has finished. I'll create a separate bug for that. 

Regarding whether the group is needed at all... Well, if you download the genitals proxies, it rather quickly becomes visible if those parts are left out of shaders. And for the shaders to work, the vertex group needs to be there.

For the other issue... the extra-everything shader is most obviously not fit for all use cases. It might make sense to split it into different shaders to cater for situations where only certain areas are relevant. For clothed characters with hair, several complex shader groups could be left out of the shader.

For creating shareable assets... That's one focus of the beta release. I hope to fix up the stuff on the "create assets" panel so that they become reasonably intuitive and easy to use to create stuff that fit in the asset repositories.

**Comment by joepal1976** [comment]
@Vidyut : I've made change in #186 for being able to preselect a vertex group. Can you check if this makes things better?

**Comment by joepal1976** [comment]
Having implemented this I later started to wonder if you were actually talking about material slot rather than vertex group. What I implemented now was the latter. But I plan to extend the functionality to also auto select the corresponding material slot if it exists.

**Comment by Vidyut** [comment]
Uh-oh. I definitely meant material slot. lol. I thought the group being selected was what made the material active.

Basically, what I mean is whenever we go to the shader view to work on a material, it should be the "body" selected ideally, because that's what needs the most edits. Eyes, lips, nails and such come after that. I've never needed to edit genital material a single time, and that's what I end up editing, because most of the time we just move to the shading editor and edit. Most of my models don't have this structure where i have different shaders on the same mesh. So I'm not in the habit of checking/switching materials before editing.

Apart from it being a possible inconvenience for experts, newbies (like me, then) might be puzzled why their edits aren't showing up on their model. I know I was puzzled for a long time until I turned the skin green or blue in frustration to see _some_ change happen, and the crotch lit up. So I checked what material I was editing and there it was, truncated out of sight, but clearly visible when selected - genitals. It is just counterintuitive.
